src/Conversion/ONNXToTOSA/Tensor/PaddingOp.cpp

```diff
@@ -0,0 +1,169 @@
+//
+//  Attributes:
+//    mode	::mlir::StringAttr	string attribute
+//
+//  Operands:
+//  data	  tensor of 8-bit/16-bit/32-bit/64-bit unsigned integer values
+//  or 	  tensor of 8-bit/16-bit/32-bit/64-bit signless integer values or
+//  tensor of bfloat16 type values or tensor of 16-bit/32-bit/64-bit 	  float
+//  values or tensor of string type values or tensor of 1-bit 	  signless
+//  integer values or tensor of complex type with 	  32-bit/64-bit float
+//  elements values or memref of any type values.
+//
+//  pads   tensor of 64-bit signless integer values or memref of
+//  	  any type values.
+//
+//  constant_value
+//  	  tensor of 8-bit/16-bit/32-bit/64-bit unsigned integer values or
+//         tensor of 8-bit/16-bit/32-bit/64-bit signless integer values or
+//         tensor of bfloat16 type values or tensor of 16-bit/32-bit/64-bit
+//         float values or tensor of string type values or tensor of 1-bit
+//         signless integer values or tensor of complex type with
+//         32-bit/64-bit float elements values or memref of any type values
+//         or none type.
+//
+// Results:
+//  output
+//         tensor of 8-bit/16-bit/32-bit/64-bit unsigned integer values or
+//         tensor of 8-bit/16-bit/32-bit/64-bit signless integer values or
+//         tensor of bfloat16 type values or tensor of 16-bit/32-bit/64-bit
+//         float values or tensor of string type values or tensor of 1-bit
+//         signless integer values or tensor of complex type with
+//         32-bit/64-bit float elements values or memref of any type values
+//         or none type.
+//
+
+namespace onnx_mlir {
+
+class ONNXPadOpLoweringToTOSA : public OpConversionPattern<ONNXPadOp> {
+public:
+  using OpConversionPattern<ONNXPadOp>::OpConversionPattern;
+  using OpAdaptor = typename ONNXPadOp::Adaptor;
+  LogicalResult matchAndRewrite(ONNXPadOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const final {
+
+    Location loc = op.getLoc();
+
+    Value data = adaptor.data();
+    Value pads = adaptor.pads();
+    Value constValue = adaptor.constant_value();
+
+    if (!(adaptor.mode() == "constant")) {
+      return rewriter.notifyMatchFailure(
+          op, "Only 'constant' mode is supported");
+    }
+
+    // creating the DenseElementsAttr using pads values.
+    DenseElementsAttr denseAttr =
+        llvm::dyn_cast_or_null<tosa::ConstOp>(pads.getDefiningOp())
+            .getValue()
+            .dyn_cast<DenseElementsAttr>();
+
+    // Reading the ONNX side pads values and store in the array.
+    llvm::SmallVector<APInt, 8> intValues;
+    bool paddingNeeded = false;
+    for (auto n : denseAttr.getValues<APInt>()) {
+      intValues.push_back(n);
+      if (!n.isZero())
+        paddingNeeded = true;
+    }
+    if (!paddingNeeded) {
+      // We do not need to represent the no-op pad in the resulting MLIR
+      rewriter.eraseOp(pads.getDefiningOp());
+      rewriter.eraseOp(constValue.getDefiningOp());
```

Comment:
I don't believe these can be safely erased as they may have multiple used (or be region arguments).